src/components/App.js

```diff
@@ -1,21 +1,41 @@
 /* eslint-disable jsx-quotes */
 /* eslint-disable no-unused-vars */
+/* eslint-disable react/no-access-state-in-setstate */
 
 import React from 'react';
 import Display from './display';
 import ButtonPanel from './button-panel';
 import calculate from '../logic/calculate';
 import '../styles/App.css';
 
-function App() {
-  return (
-    <div className='container'>
-      <div className='App' id='app-id'>
-        <Display result='0' display='display' />
-        <ButtonPanel className='button-panel' />
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    this.setState(calculate(this.state, buttonName));
+  }
```

Comment:
Instead of disabling the linter rule [**react/no-access-state-in-setstate**,](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-access-state-in-setstate.md) is it not better to use `this.state` outside of `this.setState`? 🤔 We could do something like this for example, or alternatively, we could declare a variable with `this.state` as its value and use in inside of the `this.setState`

```suggestion
  handleClick(buttonName) {
    const newState = calculate(this.state, buttonName);
    this.setState(newState);
  }
```